odoo/addons/base/ir/ir_actions.py

```diff
@@ -470,257 +479,55 @@ def _get_states(self):
-        while path:
-            step = path.pop(0)
-            field = self.env[model_name]._fields.get(step)
-            if not field:
-                return (False, None, 'Part of the expression (%s) is not recognized as a column in the model %s.' % (step, model_name))
-            ftype = field.type
-            if ftype not in ['many2one', 'int']:
-                return (False, None, 'Part of the expression (%s) is not a valid column type (is %s, should be a many2one or an int)' % (step, ftype))
-            if ftype == 'int' and path:
-                return (False, None, 'Part of the expression (%s) is an integer field that is only allowed at the end of an expression' % (step))
-            if ftype == 'many2one':
-                model_name = field.comodel_name
-        return (True, model_name, None)
-
     @api.constrains('code')
     def _check_python_code(self):
         for action in self.filtered('code'):
             msg = test_python_expr(expr=action.code.strip(), mode="exec")
             if msg:
                 raise ValidationError(msg)
 
-    @api.constrains('write_expression', 'model_id')
-    def _check_write_expression(self):
-        for record in self:
-            if record.write_expression and record.model_id:
-                correct, model_name, message = self._check_expression(record.write_expression, record.model_id)
-                if not correct:
-                    _logger.warning('Invalid expression: %s' % message)
-                    raise ValidationError(_('Incorrect Write Record Expression'))
-
-    @api.constrains('child_ids')
+    @api.constrains('multi_ids')
     def _check_recursion(self):
-        if not self._check_m2m_recursion('child_ids'):
+        if not self._check_m2m_recursion('multi_ids'):
             raise ValidationError(_('Recursion found in child server actions'))
 
     @api.onchange('model_id')
     def _onchange_model_id(self):
-        """ When changing the action base model, reset crud config
-        to ease value coherence. """
-        self.use_create = 'new'
-        self.use_write = 'current'
-        self.crud_model_id = self.model_id
-
-    @api.onchange('use_create', 'use_write', 'ref_object')
-    def _onchange_crud_config(self):
-        """ Wrapper on CRUD-type (create or write) on_change """
-        if self.state == 'object_create':
-            self._onchange_create_config()
-        elif self.state == 'object_write':
-            self._onchange_write_config()
-
-    def _onchange_create_config(self):
-        """ When changing the object_create type configuration:
-
-         - `new` and `copy_current`: crud_model_id is the same as base model
-         - `new_other`: user choose crud_model_id
-         - `copy_other`: disassemble the reference object to have its model
-         - if the target model has changed, then reset the link field that is
-           probably not correct anymore
-        """
-        crud_model_id = self.crud_model_id
```

Comment:
actually based on new specification, we just replace "crud_model_id" to  "create_model_id"
and in current master, we have "use_create" and based on this "crud_model_id" field become visible. but now that will be managed by "state(action to do)" and this "create_model_id".